### ISSUE TITLE
[BUG] Launcher crashes when scrolling through shortcuts

### ISSUE THREAD

**Issue by HunterArmstrong** [TBog/TBLauncher#27]
**Describe the bug**
Launcher crashes when scrolling through shortcuts in shortcuts quick list

**To Reproduce**
1. Go to quick list
2. Click on shortcut
3. Scroll

**Expected behavior**
The app will crash

**Screenshots**
N/A

**Smartphone (please complete the following information):**
 - Device: [e.g. Samsung S10+]
 - OS: [e.g. Android 10]
 - Version [e.g. v4.5]

**Comment by TBog** [comment]
I can see this crash in the Google Play Console
```
java.lang.NullPointerException: 
  at rocks.tbog.tblauncher.entry.ShortcutEntry.setIcons (ShortcutEntry.java:393)
  at rocks.tbog.tblauncher.entry.ShortcutEntry$AsyncSetEntryIcon.onPostExecute (ShortcutEntry.java:423)
  at rocks.tbog.tblauncher.entry.ShortcutEntry$AsyncSetEntryIcon.onPostExecute (ShortcutEntry.java:398)
  at android.os.AsyncTask.finish (AsyncTask.java:755)
  at android.os.AsyncTask.access$900 (AsyncTask.java:192)
  at android.os.AsyncTask$InternalHandler.handleMessage (AsyncTask.java:772)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8154)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
```
I fixed the above crash on commit 083b2638a8f30dd6dac8d5ad49f2f96761ff9994 but I need to understand what is happening before I close this issue.

**Comment by HunterArmstrong** [comment]
Well, what do you need to know?

**Comment by TBog** [comment]
I need to understand how you made it crash. I already asked for clarification above.
> I don't understand the first step. Do you mean:
> 
> 1. click in the quick list bar on the shortcut filter
> 2. go to the quick list customize content dialog
> 
> If you mean the 2nd then where do you click on the 2nd step? on _Preview_ or _From the favorites_ ?
> 
> How many shortcuts do you have? Maybe the crash happens when a particular shortcut is displayed?

**Comment by HunterArmstrong** [comment]
Okay. I'm sorry. I just jumped the gun to the second reply you made and didn't bother to see that you already asked what you didn't understand.

> I don't understand the first step. Do you mean:
> 1. click in the quick list bar on the shortcut filter
> 2. go to the quick list customize content dialog

I meant the quick list bar (1)

> How many shortcuts do you have? Maybe the crash happens when a particular shortcut is displayed? 

19 from one app.
48 from another.
67 Total.

(I didn't know if the amount of shortcuts from separate apps was a detail that was needed. So, I just went forward with it anyway to avoid a mishap)

**Comment by HunterArmstrong** [comment]
Hey. I never mentioned this too; but, the same type of crash happens when I have shortcuts exclusively in my search results list.

First, I would type a term in my search bar.

Then, there would be a rare occurrence where there aren't any apps or contacts in the search results feed. Just shortcuts added to the launcher

Once I scroll, the launcher crashes.

**Comment by TBog** [comment]
I can't duplicate your problem, but I have an idea why it happened.
When scrolling a new asynchronous task is generated to load the icon of the shortcut but before it finishes the shortcut view somehow gets deleted.
This should be fixed on the next release.

**Comment by HunterArmstrong** [comment]
Alright. Even though I'm kind of confused as to why this behavior would happen on TB and not KISS, I'm still crossing my fingers hoping it is easy to fix

**Comment by TBog** [comment]
I rewrote everything from KISS that had to do with shortcuts and everything about icon caching

**Comment by HunterArmstrong** [comment]
Can I ask why you did that? I'm not trying to criticize. I'm just curious.

**Comment by TBog** [comment]
I felt that the cache was not memory efficient and couldn't be customized.
The shortcuts were not working on my device the way I expected and so I changed that.
This is why I made this project. I wanted to do things my way.

**Comment by HunterArmstrong** [comment]
Okay. Cool beans.

**Comment by HunterArmstrong** [comment]
Hey, quick question. Would you consider clearing all the data on TB to be a temporary solution to the shortcut problem?

**Comment by TBog** [comment]
Not sure. Is the latest release on GitHub not fixing your problem?
I currently have a problem with my SSD and can't work on the project, waiting for a new drive.

**Comment by HunterArmstrong** [comment]
> Not sure. Is the latest release on GitHub not fixing your problem?

I just tried resetting it, doesn't work.

The most recent release I have is from the Google Play Store when you updated TB. Are you trying to say that there's a newer version on GitHub? If so, how do I access it?

> I currently have a problem with my SSD and can't work on the project, waiting for a new drive. 

Sorry that you have that setback. Hope a new SSD comes soon.

**Comment by TBog** [comment]
> Are you trying to say that there's a newer version on GitHub? If so, how do I access it?

https://github.com/TBog/TBLauncher/releases

**Comment by HunterArmstrong** [comment]
Downloaded it. How do I adjust the text size for the search results?

**Comment by TBog** [comment]
TBLauncher Settings ➡ User Interface ➡ Search bar ➡ Bar and text size

**Comment by HunterArmstrong** [comment]
> TBLauncher Settings ➡ User Interface ➡ Search bar ➡ Bar and text size

This Setting only works for the text in the search bar. When I asked about changing the size of the text, I meant the text in the search *results*. Y'know the apps, shortcuts, contacts, etc.

**Comment by HunterArmstrong** [comment]
Just saw your latest commit. I'll let you do your thing.

**Comment by HunterArmstrong** [comment]
Fixed in 4.6
